otter/integration/lib/autoscale.py

```diff
@@ -262,17 +280,58 @@ def record_results(resp):
             return rcs
 
         return (
-            treq.post(
+            self.treq.post(
                 "%s/groups" % str(rcs.endpoints["otter"]),
                 json.dumps(self.group_config),
                 headers=headers(str(rcs.token)),
                 pool=self.pool
             )
             .addCallback(check_success, [201])
-            .addCallback(treq.json_content)
+            .addCallback(self.treq.json_content)
             .addCallback(record_results)
         )
 
+    def get_servicenet_ips(self, rcs, server_ids=None):
+        """
+        Gets the servicenet IPs for the following server IDs - if no IDs are
+        provided, gets the servicenet IPs for all the active servers on the
+        group.
+
+        :param iterable server_ids: An iterable of server ids - this function
+            does not check whether the servers belong the the group
+
+        :return: A mapping of server ID to servicenet address
+        :rtype: ``dict``
+        """
+        def _extract_servicenet_id(server_info):
+            private = server_info['server']['addresses'].get('private')
+            if private is not None:
+                return [addr['addr'] for addr in private
+                        if addr['version'] == 4][0]
+            return None
+
+        def _get_the_ips(the_server_ids):
+            the_server_ids = set(the_server_ids)
+            return gatherResults([
+                self.treq.get(
+                    "{0}/servers/{1}".format(rcs.endpoints["nova"], server_id),
```

Comment:
How would you feel about this function also checking if the endpoint is there, and if not, then finding it?